### DISCUSSION TITLE
copilot.vim: Error when opening Neovim

### DISCUSSION THREAD

**Discussion by yuntan** [community/community#124358] [Copilot Conversations]
### Select Topic Area

Bug

### Body

The following error occurs when opening Neovim.

```
VimEnter Autocommands for "*"..function <SNR>91_MapTab の処理中にエラーが検出されました:
行   15:
E488: Trailing characters: SNR>133_IsAnEmptyQuote() ? '<C-O>:call <SNR>133_Indent(1)<CR>' : '<Tab>' }):  <SNR>133_IsAnEmptyQuote() ? '<C-O>:call <SNR>133_Indent(1)<CR>' : '<Tab>' })
```

copilot.vim: v1.31.0

Neovim version:

```
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1713773202
```

Edit: I upgraded Neovim to v0.10.0, but the problem still exists.

Edit: Sorry, this seems to be a bug of gabrielelana/vim-markdown.

Edit: Enabling vim-markdown and copilot.vim and opening a markdown file caused the problem. `<Tab>` mapping fallback functionary of `MapTab()` seems to make wrong expression. Sharing the same mapping between distinct plugins is too difficult...

Edit: The results of `:autocmd` are followed.
https://gist.github.com/yuntan/3e154811d3d991532123d1f669ce2ce3

**Comment by shaunemslie** [comment]
Are you still having this issue?

**Reply by yuntan** [reply]
Yes, I'm having the issue on copilot.vim v1.34.0.

**Reply by shaunemslie** [reply]
Strangely, I'm using `preservim/vim-markdown`, and it seems to work fine. I'll try to help out though, could I get the results of `:autocmd` if possible?

**Reply by yuntan** [reply]
I've uploaded the results of `:autocmd` (too long...).
https://gist.github.com/yuntan/3e154811d3d991532123d1f669ce2ce3

**Reply by shaunemslie** [reply]
I suspect the issue is in the following code:
```vim
    if tab_map.expr
      let tab_fallback = '{ -> ' . tab_map.rhs . ' }'
    else
      let tab_fallback = substitute(json_encode(tab_map.rhs), '<', '\\<', 'g')
    endif
```

I've just noted that you mentioned `gabrielelana/vim-markdown` and I mistook this for `preservim/vim-markdown` in my previous comment. I'll investigate this plugin and see what I can do, my best guess at the moment is somewhere it maps the `<Tab>` key to something that contains a `<` or `>` character in exclusion to using them as an `expr`.

**Reply by shaunemslie** [reply]
This right here would be the issue, gabrielelana/vim-markdown/ftplugin/markdown.vim (l.162):
```vim
  if g:markdown_enable_insert_mode_mappings
    " Indenting things
    inoremap <silent> <buffer> <script> <expr> <Tab>
      \ <SID>IsAnEmptyListItem() \|\| <SID>IsAnEmptyQuote() ? '<C-O>:call <SID>Indent(1)<CR>' : '<Tab>'
    inoremap <silent> <buffer> <script> <expr> <S-Tab>
      \ <SID>IsAnEmptyListItem() \|\| <SID>IsAnEmptyQuote() ? '<C-O>:call <SID>Indent(0)<CR>' : '<Tab>'
```

The `gabrielelana/vim-markdown` repository seems stale, the last commit being 4 years ago. I'd fork it and attempt to fix, otherwise find an active replacement.

**Comment by shaunemslie** [comment]
gabrielelana/vim-markdown/ftplugin/markdown.vim (l.141):
```vim
function! s:IsAnEmptyQuote()
  return getline('.') =~ '\v^\s*(\s?\>)+\s*$'
endfunction
```
